hollow-diff-ui/src/main/java/com/netflix/hollow/history/ui/jetty/HollowHistoryUIServer.java

```diff
@@ -49,7 +53,7 @@ public HollowHistoryUIServer(HollowHistory history, int port) {
     }
 
     public HollowHistoryUIServer(HollowHistoryUI ui, int port) {
-        this.server = FACTORY.newServer(ui, port);
+        this.server = new JettyBasedUIServer(ui, port);
```

Comment:
We will have two classes with same implementations but different name to avoid breaking any applications using Jetty based class names. The implementation itself does not use jetty. The newer class name is UIWebServer.
e.g:
hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/jetty/JettyBasedUIServer.java
hollow-diff-ui/src/main/java/com/netflix/hollow/diff/ui/webserver/UIWebServer.java

Reply:
* +1 to keeping the old class around, marking as @deprecated with a note to use the new class instead
* Here we can instantiate the new class (so we stop using the old class internally)
* The implementation of the old class can be changed so that it is a wrapper around the new class (instead of duplicating code).

Reply:
Already added deprecated.

Reply:
Since new server class is not public. The parallel class cannot be accessed outside package, so instantiate the new class may not be an option.

Reply:
I tried a different approach by extending from webserver class. Let me know if its reasonable.

Reply:
In this case, I think composition makes more sense because it isn't really a subclass-superclass relationship.  Moreover composition is favorable over inheritance in general for more flexibility.

Reply:
changed to composition